### PR TITLE
Adding aws_dms_event_subscription, aws_dms_replication_instance, aws_dms_replication_task to v1beta1 version

### LINKED ISSUE #31: Moving dms (3) resources to v1beta1 version
Moving 3 resources in the `dms` group:
- [x] [`aws_dms_event_subscription`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/dms_event_subscription)
- [x] [`aws_dms_replication_instance`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/dms_replication_instance)
- [x] [`aws_dms_replication_task`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/dms_replication_task)

Extracted from https://github.com/upbound/official-providers/issues/252

### PR DESCRIPTION
**Pull request by mykolalosev**
### Description of your changes
Adding 3 resources in the `dms` group:
- [x] [`aws_dms_event_subscription`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/dms_event_subscription)
- [x] [`aws_dms_replication_instance`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/dms_replication_instance)
- [x] [`aws_dms_replication_task`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/dms_replication_task)

Fixes #31 

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested
Manual:
aws_dms_event_subscription - required Endpoint resources need a valid certificateArn and kmsKeyArn values.
aws_dms_event_subscription

aws_dms_replication_task - required Endpoint resources need a valid certificateArn and kmsKeyArn values.
aws_dms_replication_task

Manual & Uptest:
aws_dms_replication_instance - https://github.com/upbound/provider-aws/actions/runs/3893158413

### CODE REVIEW

**Comment by mykolalosev** [comment]
/test-examples="examples/dms/replicationinstance.yaml"

File: config/provider.go [hunk 1]
```diff
@@ -208,7 +208,6 @@ func GetProvider() *config.Provider {
 		s3.Configure,
 		secretsmanager.Configure,
 		servicecatalog.Configure,
-		devicefarm.Configure,
```
Comment (sergenyalcin) [inline config/provider.go:211]: I think you removed this statement because it is a duplicate, right?

Comment (mykolalosev) [reply config/provider.go:211]: Yes, that's right.

**Review by sergenyalcin — APPROVED** [review]
Thanks @mykolalosev LGTM!

**Pull request merged**